### PR TITLE
Feat/implementing trigger for popper

### PR DESCRIPTION
**Pull request by jw-foss**
- Add `trigger` API for `ElPopper`
- Add arguments for `ClickOutside`

### CODE REVIEW

File: packages/tooltip/src/index.vue [hunk 1]
```diff
@@ -87,6 +87,10 @@ export default defineComponent({
       type: String,
       default: 'el-fade-in-linear',
     },
+    trigger: {
+      type: [String, Array] as PropType<string | string[]>,
+      default: () => ['hover'],
+    },
```
Comment (JeremyWuuuuu) [inline packages/tooltip/src/index.vue:93]: I think I should set a specific type here

File: packages/directives/click-outside/index.ts [hunk 2]
```diff
@@ -30,21 +30,44 @@ function createDocumentHandler(
   el: HTMLElement,
   binding: DirectiveBinding,
 ): DocumentHandler {
+  let excludes: HTMLElement[] = []
+  if (Array.isArray(binding.arg)) {
+    excludes = binding.arg
+  } else {
+    // due to current implementation on binding type is wrong the type casting is necessary here
+    excludes.push(binding.arg as unknown as HTMLElement)
```
Comment (zouhangwithsweet) [inline packages/directives/click-outside/index.ts:38]: `undefined` cannot be pushed in `excludes`

File: packages/popper/src/index.vue [hunk 3]
```diff
@@ -153,31 +153,7 @@ export default defineComponent({
...
-      popperId,
-      doDestroy,
-      onShow,
-      onHide,
-      popperRef,
-      popperInstance,
-      initializePopper,
-      visible,
-    }
+    return usePopper(props as IPopperOptions)
```
Comment (zouhangwithsweet) [inline packages/popper/src/index.vue:156]: The type of the return value of this function should be correctly deduced

**Review by zouhangwithsweet — APPROVED** [review]

File: packages/directives/click-outside/index.ts (hunk 2 above)
Comment (JeremyWuuuuu) [reply packages/directives/click-outside/index.ts:38]: Right, I will add a type check on this

**Pull request merged**